### PR TITLE
[LTC] Implement Gandalf, Westward Voyager

### PR DESCRIPTION
**Pull request by Susucre**
There is a big commit to refactor `SpellCastControllerTriggeredAbility` before-end:
- cleaning the constructors, with a few static create for convenience.
- Replacing the weird double boolean by an optional SetTargetPointer argument.

### CODE REVIEW

**Comment by Susucre** [comment]
[[Gandalf, Westward Voyager]]

File: Mage/src/main/java/mage/abilities/common/SpellCastControllerTriggeredAbility.java [hunk 1]
```diff
@@ -86,8 +88,13 @@ public boolean checkTrigger(GameEvent event, Game game) {
...
         this.getEffects().setValue("spellCast", spell);
-        if (rememberSource) {
-            this.getEffects().setTargetPointer(new FixedTarget(rememberSourceAsCard ? spell.getCard().getId() : spell.getId(), game));
+        switch (setTargetPointer) {
+            case SPELL:
+                getEffects().setTargetPointer(new FixedTarget(spell.getId(), game));
+                break;
+            case CARD:
+                getEffects().setTargetPointer(new FixedTarget(spell.getCard().getId()));
+                break;
```
Comment (xenohedron) [inline Mage/src/main/java/mage/abilities/common/SpellCastControllerTriggeredAbility.java:97]: Might be better to make explicit with case NONE: break, and default: throw UnsupportedOperationException

File: Mage.Sets/src/mage/cards/g/GandalfWestwardVoyager.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        super(card);
+    }
+
+    @Override
+    public GandalfWestwardVoyager copy() {
+        return new GandalfWestwardVoyager(this);
+    }
+}
+
+class GandalWestwardVoyagerEffect extends OneShotEffect {
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/g/GandalfWestwardVoyager.java:58]: correct spelling to Gandalf

**Review by xenohedron — COMMENTED** [review]
Nice cleanup, looks good, minor nits

**Review by xenohedron — APPROVED** [review]

File: Mage.Sets/src/mage/cards/t/TuvasaTheSunlit.java [hunk 3]
```diff
@@ -76,9 +70,15 @@ public TuvasaTheSunlit copy() {
 
 class TuvasaTheSunlitTriggeredAbility extends SpellCastControllerTriggeredAbility {
 
+    private static final FilterSpell filter = new FilterSpell("an enchantment spell");
```
[the remark this answers is not in the document]
Comment (xenohedron) [reply Mage.Sets/src/mage/cards/t/TuvasaTheSunlit.java:72]: fixed in 2ec4401

**Pull request merged**